test/archethic/mining/validation_context_test.exs

```diff
@@ -158,6 +159,83 @@ defmodule Archethic.Mining.ValidationContextTest do
                }
              } = ValidationContext.create_validation_stamp(validation_context)
     end
+
+    test "should handle the MUCO correctly" do
+      timestamp = DateTime.utc_now() |> DateTime.truncate(:millisecond)
+      transfer_address = random_address()
+      resolved_address = random_address()
+      muco_addr1 = random_address()
+      muco_addr2 = random_address()
+
+      start_supervised!(RewardTokens)
+      RewardTokens.add_reward_token_address(muco_addr1)
+      RewardTokens.add_reward_token_address(muco_addr2)
+
+      validation_context = %ValidationContext{
+        create_context(timestamp,
+          utxos: [
+            %UnspentOutput{
+              from: random_address(),
+              amount: 10,
+              type: {:token, muco_addr1, 0},
+              timestamp: timestamp
+            },
+            %UnspentOutput{
+              from: random_address(),
+              amount: 10,
+              type: {:token, muco_addr2, 0},
+              timestamp: timestamp
+            }
+          ]
+        )
+        | transaction:
+            Transaction.new(
```

Comment:
You should use the function `TransactionFactory.create_non_valided_transaction` and pass it in the opts field of the create_context function as you did for utxo